walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,125 @@
+            if (intersection.size() > 0) {
+                mandatoryAttendeesUnavailableTimes.add(event.getWhen());
+                optionalAttendeesUnavailableTimes.add(event.getWhen());
+            }
+            intersection = new HashSet<String>();
+            intersection.addAll(event.getAttendees());
+            intersection.retainAll(request.getOptionalAttendees());
+            //if an optional attendee has an event
+            if (intersection.size() > 0) {
+                optionalAttendeesUnavailableTimes.add(event.getWhen());
+            }
+        }
+        //the time slots that mandatory attendees are available
+        ArrayList<TimeRange> mandatoryAttendeesAvailableTimes = findAvailableTimes(mandatoryAttendeesUnavailableTimes, request);
+        //the time slots that mandatory and optional attendees are available
+        ArrayList<TimeRange> optionalAttendeesAvailableTimes = findAvailableTimes(optionalAttendeesUnavailableTimes, request);
+        //if there are time slots that mandatory and optional attendees are available
+        if (optionalAttendeesAvailableTimes.size() > 0) {
+            return optionalAttendeesAvailableTimes;
+        }
+        else {
+            return mandatoryAttendeesAvailableTimes;
+        }
+    }
+
+    private ArrayList<TimeRange> findAvailableTimes(ArrayList<TimeRange> unavailableTimes, MeetingRequest request) {
+        //sort time ranges by start time
+        TimeRangeComparator comparator = new TimeRangeComparator();
+        unavailableTimes.sort(comparator);
+        //combine overlapping time ranges
+        int i = 0;
+        while (i < unavailableTimes.size() - 1) {
+            if (unavailableTimes.get(i).overlaps(unavailableTimes.get(i + 1)) == true) {
+                int start = unavailableTimes.get(i).start();
+                int end = (Long.compare(unavailableTimes.get(i).end(), unavailableTimes.get(i + 1).end()) > 0) ? unavailableTimes.get(i).end() : unavailableTimes.get(i + 1).end();
+                unavailableTimes.set(i, TimeRange.fromStartEnd(start, end, false));
+                unavailableTimes.remove(i + 1);
+            }
+            else {
+                i++;
+            }
+        }
+        //the time slots that attendees are available
+        ArrayList<TimeRange> availableTimes = new ArrayList<TimeRange>();
+        //if there are no time slots that attendees are unavailable, the whole day is available
+        if (unavailableTimes.size() == 0) {
+            int start;
+            int end;
+            TimeRange time;
+            start = TimeRange.START_OF_DAY;
+            end = TimeRange.END_OF_DAY;
+            time = TimeRange.fromStartEnd(start, end, true);
+            if (time.duration() >= request.getDuration()) {
+                availableTimes.add(time);
+            }
+        }
+        //if there are time slots that attendees are unavailable, the gaps in the schedules are available
+        else if (unavailableTimes.size() == 1) {
```

Comment:
the else if belongs with the } on line 87
https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style

that comment can be the first line inside the else if body.